### DISCUSSION TITLE
Shortcut for external viewer

### DISCUSSION THREAD

**Discussion by stphntlbt** [sumatrapdfreader/sumatrapdf#2368] [Ideas]
Hello, can you add an option for **keyboard shortcut** to **run external viewers** (maybe with `SHIFT + 1` (the first viewer, detected or declared in the settings), `SHIFT+2` (the second, detected or declared in the settings)... 
Would be great (Sumatra as the main viewer, and externals for editing with direct access)
Thanks

**Comment by GitHubRulesOK** [comment]
You set your own shortcut in settings so the standard ones are <kbd>Alt</kbd><kbd>F</kbd><kbd>M</kbd> etc. but numbers are already allocated to the 10 files however you can add say <kbd>Alt</kbd><kbd>F</kbd><kbd>#</kbd> or any other unused.

I agree the list of free keys is increasingly limited but often one is suitable for calling multiple viewers so I may shell out to a set of icons to pick Edge, Tracker, Acrobat or Sejda or, or, limited by my ability to add an iconic picker. see https://www.lerup.com/LaunchBar/ Note not all can accept a page number like acrobat or edge (but most will) however all will accept the current filename.
![image](https://user-images.githubusercontent.com/12754913/155167297-d335f696-e0ed-4b52-919b-bee3fd19b848.png)

you can have a different set for each file type so another set on same or different shortcut could include ebook handlers etc.

**Reply by stphntlbt** [reply]
Thanks, but there's no way to add or edit existing shortcuts (no menu, no group in the settings files or even in the documentation).
I bypassed this by using RoboTask Lite, a free trigger/action software that sends keys to sumatra (CTRL + SHIFT + A = ALT + F > A)

**Reply by GitHubRulesOK** [reply]
You can use any keyboard automation such as AutoHot Key but in SumatraPDF just add & (ampersand) to your custom call so this one is  ALT+F+<kbd>L</kbd> to delete **L**ocked files **L**ater

	[
		CommandLine = C:\windows\system32\cmd.exe /r echo del "%1" >>"%USERPROFILE%\Desktop\DeletionList.cmd"
		Name = send filename to a Delete &Later List
	]

PS ALT+F+A is AFAIK call installed Acrobat ? F for Foxit etc

**Reply by stphntlbt** [reply]
Thanks for the `&` tips!

**Comment by GitHubRulesOK** [comment]
just an update to say 3.6 has the ability to set a KEY for any external command so for an example of running an HTA see https://github.com/GitHubRulesOK/SumatraPDF-Plus?tab=readme-ov-file#piccolohta note 3.6.### prerelease will be needed for that most recent example of an internal colour picker HOWEVER the principle is the same in 3.6 release to use any key as an external viewer trigger.
